core/state.go

```diff
@@ -29,43 +29,47 @@ package core
 
 import (
 	"math/rand"
-	"time"
 	"sync"
+	"time"
 
+	"context"
+	"encoding/json"
 	"github.com/AliceO2Group/Control/core/confsys"
+	"github.com/AliceO2Group/Control/core/controlcommands"
+	"github.com/AliceO2Group/Control/core/environment"
+	"github.com/AliceO2Group/Control/core/task"
+	"github.com/looplab/fsm"
 	"github.com/mesos/mesos-go/api/v1/lib"
 	"github.com/mesos/mesos-go/api/v1/lib/backoff"
 	"github.com/mesos/mesos-go/api/v1/lib/scheduler/calls"
-	"github.com/looplab/fsm"
-	"github.com/AliceO2Group/Control/core/environment"
-	"encoding/json"
-	"github.com/AliceO2Group/Control/core/controlcommands"
-	"context"
-	"github.com/AliceO2Group/Control/core/task"
+
+	"github.com/spf13/viper"
 )
 
-func newInternalState(cfg Config, shutdown func()) (*internalState, error) {
-	metricsAPI := initMetrics(cfg)
+func newInternalState(shutdown func()) (*internalState, error) {
+	metricsAPI := initMetrics()
 	executorInfo, err := prepareExecutorInfo(
-		cfg.executor,
-		cfg.mesosExecutorImage,
-		buildWantsExecutorResources(cfg),
-		cfg.mesosJobRestartDelay,
+		viper.GetString("executor"),
+		viper.GetString("mesosExecutorImage"),
+		buildWantsExecutorResources(viper.GetFloat64("executorCPU"),
+			viper.GetFloat64("executorMemory")),
```

Comment:
This makes me think that in hindsight it might have been nice to have a strongly typed structure to unmarshal into, but we can leave it like this for the time being. Either way most of these values are only used in a few spots.